packages/features/ee/billing/teams/internal-team-billing.test.ts

```diff
@@ -0,0 +1,155 @@
+import prismaMock from "../../../../../tests/libs/__mocks__/prismaMock";
+
+import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
+
+import { purchaseTeamOrOrgSubscription } from "@calcom/features/ee/teams/lib/payments";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { Redirect } from "@calcom/lib/redirect";
+
+import * as billingModule from "..";
+import { InternalTeamBilling } from "./internal-team-billing";
+
+vi.mock("@calcom/lib/constants", async () => {
+  const actual = await vi.importActual("@calcom/lib/constants");
+  return {
+    ...actual,
+    WEBAPP_URL: "http://localhost:3000",
+  };
+});
+
+vi.mock("..", () => ({
+  default: {
+    handleSubscriptionCancel: vi.fn(),
+    handleSubscriptionUpdate: vi.fn(),
+    checkoutSessionIsPaid: vi.fn(),
+  },
+}));
+
+vi.mock("@calcom/features/ee/teams/lib/payments", () => ({
+  purchaseTeamOrOrgSubscription: vi.fn(),
+}));
+const mockTeam = {
+  id: 1,
+  metadata: {
+    subscriptionId: "sub_123",
+    subscriptionItemId: "si_456",
+    paymentId: "cs_789",
+  },
+  isOrganization: true,
+  parentId: null,
+};
+
+describe("InternalTeamBilling", () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  describe("cancel", () => {
+    const internalTeamBilling = new InternalTeamBilling(mockTeam);
```

Comment:
What does "Internal" mean here? Is it to describe that this class is handling most of the core logic for team billing? Looking at the class, seems like `TeamBillingService` would be more fitting since it's doing repository calls and logical changes. i.e. It's orchestrating like an Application Service.

Reply:
By internal. It means Cal.com internal billing.